### PR TITLE
Add self update

### PR DESCRIPTION
**Pull request by rafael-vitor**
**Summary**

closes: #1139 
Putting back `yarn self-update` as we now have a way of knowing which installation method was used and how to properly update. 

**Test plan**

As this is my first contribution to this project I'd really appreciate some feedback before going into implementing tests. I'll add them asap.  

I'm looking forward for some feedback on this! Thanks!

### CODE REVIEW

**Comment by AlexWayfer** [comment]
We have `upgrade` command for packages, not `update`, and probably should have `self-upgrade` instead of `self-update`. Also, in `apt` there are `update` for repositories refresh and `upgrade` for packages update.

**Comment by arcanis** [comment]
I believe this feature conflicts in spirit with #6673, which is already in master

**Comment by rafael-vitor** [comment]
@arcanis I see. Many ppl still refer to #1139 to try and find the way to properly upgrade yarn as the docs still mention it. 

Do you think there's room for both commands? Maybe self-upgrade is worth keeping for the clarity of its name. Maybe even as an alias to `yarn policies set-version`. 

Otherwise, we should work on making the community aware of this by linking to it on the current `self-update` doc and/or removing it eventually.

File: src/cli/commands/self-update.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    await execCommand({
+      // Could not figure out to which stage should this belong to.
+      stage: 'self-update',
+      config,
+      cmd: command,
+      cwd: config.cwd,
+      isInteractive: true,
+    });
+  } else {
+    reporter.info('could not find command for instalation method');
```
Comment (bf4) [inline src/cli/commands/self-update.js]: typo

**Pull request closed without merging**